app/scripts/models/reliers/oauth.js

```diff
@@ -204,6 +207,42 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Check if the relier wants access to the account encryption keys.
+     *
+     * @returns {Boolean}
+     */
+    wantsKeys () {
+      return !! (this.has('keysJwk') && this._config.scopedKeysEnabled && this._validateKeyScope());
+    },
+
+    /**
+     * Validate the requested scope with the relier redirect uri.
+     * At least one valid match must be found to successfully validate
+     * @returns {boolean}
+     * @private
+     */
+    _validateKeyScope () {
+      const validation = this._config.scopedKeysValidation || {};
+      let foundRedirectScopeMatch = false;
+
+      if (! this.get('scope')) {
+        throw new Error('Invalid scope parameter');
+      }
+
+      scopeStrToArray(this.get('scope')).forEach((scope) => {
+        if (validation.hasOwnProperty(scope)) {
```

Comment:
If a scope is not listed in [the big list](https://github.com/mozilla/fxa-content-server/pull/5675/files#diff-a7802dc9db070ba75283c9fc3199eb02R450) is it silently ignored? Or, can keys be requested for the non-existent scope? Or will there be an error before this point?

Reply:
IIUC, scopes that are not in the big list are ignored here, but if you didn't end up specifying *any* scopes from that list, then you'll get an error due to the check on `foundRedirectScopeMatch` below.